### ISSUE TITLE
Better in-editor coverage reporting after running tests

### ISSUE THREAD

**Issue by daniel-v** [flutter/flutter#60494]
## Use case

I like tests! :) I also enjoy pretty colors next to my code lines that tells me: "well done, you covered so much of this file!" or quite the opposite sometimes. Basically, I'm looking for a **developer experience**, where I get to draft some tests and go from there with the good ol' red-green-refactor cycle. Test coverage is an important piece of it as it tells me if something skipped my mind.

I recently asked the question for Flutter Day if it was possible to get this kind of feedback. If you want, you can see the @filiph 's answer here: https://youtu.be/cG1VDcCwOLE?t=2336

In short, the answer is no (with some caveats). 

## Proposal

- we can collect lcov info with current tooling
- editors and IDEs already have the capability of visualizing line coverage

Can we somehow connect these so that I get to see the report after running tests? I myself use Android Studio, not sure if it is an existing issue for VSCode too or not.

**Comment by iapicca** [comment]
Hi @daniel-v 
please consider this issue purely related to the "flutter side" of your proposal; 
ide implementations should be addressed in the dedicate github repo
([android studio](https://github.com/flutter/flutter-intellij/issues), [vs code](https://github.com/Dart-Code/Dart-Code/issues))
thank you for your contribution

**Comment by daniel-v** [comment]
Thank you @iapicca !

Totally get it!
It seems, this issue has come up a few times for intellij-flutter plugin. I'll just list them for easier navigation:

https://github.com/flutter/flutter-intellij/issues/3860
https://github.com/flutter/flutter-intellij/issues/718
https://github.com/flutter/flutter-intellij/issues/581

I'm a lot less invested in VS Code but if anyone would want to help with that, that'd be great!

**Comment by jonahwilliams** [comment]
lcov is an extremely well known format and there are tons of tools for it. I use Coverage Gutters (ryanluker.vscode-coverage-gutters) in vscode which has an explicit watch mode.

Closing as this is not a feature that can or should be built by the flutter team

**Comment by daniel-v** [comment]
@jonahwilliams Thank you for the tip on what you are using, I'll take a look! 

However, I do not think it's justified to close this issue. Flutter, in itself, is way more than a simple set of APIs. It is an entire ecosystem for writing mobile/desktop/web apps. Part of that ecosystem is the developer experience. I know it's an important aspect to the Flutter team; they already pushed features that make it loved among devs, such as hot reload or an excellent CLI.

Another area, where Flutter could excel is it's support for testing, not only on a CI server but also during development, it's integration with IDEs and so on. Even if this issue remains closed, I strongly believe that these concerns should be on the roadmap of the Flutter team as well.
